src/OpenTelemetry/Batch.cs

```diff
@@ -140,6 +145,29 @@ public struct Enumerator : IEnumerator<T>
                 return false;
             };
 
+            private static readonly BatchEnumeratorMoveNextFunc MoveNextCircularBufferLogRecord = (ref Enumerator enumerator) =>
+            {
+                var circularBuffer = enumerator.circularBuffer;
+
+                var currentItem = enumerator.Current;
+                if (currentItem != null)
+                {
+                    if (typeof(T) == typeof(LogRecord))
```

Comment:
I see `typeof(T) == typeof(LogRecord)` repeated in a few places. The others are Dispose method or ctor, but this one in the MoveNext seems like it would get hit often.

Consider moving this to a `private readonly bool IsLogRecord`.

Reply:
See my comment above. All of these type checks are strategic for the JIT to optimize them away.